### PR TITLE
fix: unrug cosmos SDK rewards

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

ATOM balances are currently extremely inflated, as rewards are extremely high, and not matching the actual on-chain rewards.
The reason why this happens is because of our previous assumption that validators should yield rewards denominated in the same asset - which for all intents and purposes *should* be the case since we're only supporting Cosmos SDK native staking, which yields rewards in uatom (read: ATOM).

However, this isn't what's actually returned by unchained currently. See the current `/api/v1/account` call for the demo wallet Cosmos account:

https://paw.pt/h7HUCAWc

We're getting rewards in a bunch of different denoms:

image

These denoms are from various assets such as [staked NTRN from Neutron](https://github.com/cosmostation/chainlist/blob/348bc9297da9e6b1acda9d0569bf9b8516a2b711/chain/cosmos/assets.json#L736), [staked stOSMO from Stride], etc. Regardless of whether validators legitimately yield these rewards, this is not something we actually support, hence this PR adds some filter to ensure we're only dealing with same-denom rewards.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/5201

## Risk

None

## Testing

- Cosmos SDK rewards balances look sane, and so does the total balance as a result

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

### Prod

image

### This diff

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5213** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5213" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5213?utm_source=stack-comment).

File: src/state/migrations/index.ts [hunk 1]
```diff
@@ -7,4 +7,5 @@ export const migrations = {
   1: clearOpportunities,
   2: clearPortfolio,
   3: clearOpportunities,
+  4: clearOpportunities,
```
Comment (woodenfurniture) [inline src/state/migrations/index.ts:10]: see comment above line 4 🙂
`// when adding a migration, dont forget to increment `persistConfig.version` in `src/state/store.js``

**Review by woodenfurniture — CHANGES_REQUESTED** [review]
tiny change required to run the migration

File: src/state/migrations/index.ts (hunk 1 above)
Comment (woodenfurniture) [reply src/state/migrations/index.ts:10]: hell yes

**Review by woodenfurniture — APPROVED** [review]

**Pull request merged**